### PR TITLE
Merge openshift-sdn into origin

### PR DESCRIPTION
**Pull request by danwinship**
Background: openshift-sdn was originally a standalone program, but that stopped being true a long time ago. These days it can't be compiled on its own, can't be tested on its own, etc, and yet it is still a separate git module for historical reasons. (There was also a period of time when we thought that openshift-sdn was eventually going to go away [and be replaced with flannel], so we didn't want to move it into origin just to later get rid of it. But that's not the plan any more.)

So, this gets rid of openshift-sdn as a dependency of origin and moves the code directly into the origin tree instead:
- openshift-sdn/plugins/osdn → origin/pkg/sdn/plugin
- openshift-sdn/pkg/cmd/admin/network → origin/pkg/cmd/admin/network
- openshift-sdn/pkg/{ipcmd,ovs,netutils} → origin/pkg/util/{ipcmd,ovs,netutils}

(That leaves hack/debug.sh still in the old openshift-sdn module, and it needs to be figured out later, or maybe it will just stay in openshift-sdn until we manage to migrate all the code into the origin diagnostic tool.)

The commits just move the code directly from Godeps/ to the new locations, so that the files will at least still keep the history of imports from openshift-sdn (with commit messages that contain the relevant openshift-sdn commit hashes) rather than losing all history. Unfortunately, `hack/verify-upstream-commits.sh` doesn't like us moving things around like this. Maybe we can just ignore it?

@openshift/networking PTAL. This is built on top of #9476 so ignore that part.

### CODE REVIEW

**Comment by eparis** [comment]
I THINK the bot can still work....   But if you do it that way and we have to use the big green button we'll figure that out....

**Comment by smarterclayton** [comment]
If you provide a ready commit i can do that.

**Comment by pravisankar** [comment]
[test]

**Comment by smarterclayton** [comment]
I would prefer to keep attribution, so let me know when you guys are in a
shape for me to do a subtree merge

On Thu, Jun 23, 2016 at 3:42 PM, Ravi Sankar Penta <notifications@github.com

**Comment by liggitt** [comment]
I think deleting the godep followed by a subtree merge into the right place in the repo would be better than disabling the upstream check

**Comment by danwinship** [comment]
Yeah, that's what I did. The problem is that it's considering the entire history of openshift-sdn/Godeps to be new commits, and insists that they all need to conform, even though the directory gets deleted in the end.

I just disabled the check for now to let the rest of travis run. Once this actually gets merged it won't matter any more since the check only gets run on commits that haven't been merged to master yet.

**Comment by danwinship** [comment]
[test]

**Comment by danwinship** [comment]
ok, repushed without the "temporarily disable verify-upstream-commits.sh", so travis is going to fail again. (But it passed before.) The last commit on the branch is new ("Make former openshift-sdn code pass govet")

**Comment by dcbw** [comment]
LGTM

**Comment by smarterclayton** [comment]
@liggitt can you do final pass on this?

**Comment by liggitt** [comment]
will look once https://github.com/openshift/origin/pull/9476/ merges and this is rebased... tough to follow at the moment

**Comment by danwinship** [comment]
[test]
(verified that "make verify" passes other than verify-upstream-commits.sh)

**Comment by danwinship** [comment]
[test]

(verified that "make verify" passes other than verify-upstream-commits.sh)

**Comment by danwinship** [comment]
@liggitt this is now rebased and ready for final review

**Comment by liggitt** [comment]
looks like the package names in files need to change (`osdn` changed to `plugin`, but the files still say `pkg osdn`)

File: pkg/util/ovs/ovs.go [hunk 1]
```diff
@@ -5,9 +5,19 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/openshift/openshift-sdn/pkg/exec"
+	"k8s.io/kubernetes/pkg/util/exec"
 )
 
+var execer exec.Interface
```
Comment (liggitt) [inline pkg/util/ovs/ovs.go]: is there a way to do this without a global variable?

Comment (danwinship) [inline pkg/util/ovs/ovs.go]: We could do it like kubernetes does, and make ovs.NewTransaction() take an exec.Interface argument, and then real callers would do `otx := ovs.NewTransaction(kexec.New(), "br0")`, and the test code would do `otx := ovs.NewTransaction(&fakeExec, "br0")`

Comment (liggitt) [inline pkg/util/ovs/ovs.go]: that seems better

**Comment by danwinship** [comment]
[test]

**Comment by danwinship** [comment]
flake #9490 
[test]

File: pkg/sdn/plugin/controller.go [hunk 2]
```diff
@@ -50,7 +51,8 @@ func getPluginVersion(multitenant bool) []string {
 func alreadySetUp(multitenant bool, localSubnetGatewayCIDR string) bool {
 	var found bool
 
-	itx := ipcmd.NewTransaction(LBR)
+	kexec := kexec.New()
```
Comment (liggitt) [inline pkg/sdn/plugin/controller.go]: overwriting the package name with an instance of the object makes it really hard to follow code using this

**Comment by liggitt** [comment]
nit on `kexec` package overwriting, LGTM otherwise

**Comment by smarterclayton** [comment]
This is allowed to merge before the rebase.

File: pkg/sdn/plugin/controller.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/sdn/plugin/controller.go]: You can do this in a follow up (link it here once you're done)

**Comment by smarterclayton** [comment]
LGTM [merge] with follow up for nit.

**Comment by smarterclayton** [comment]
LGTM [merge] with nit followup
